### ISSUE TITLE
lock files always inconsistent when using centrally managed packages

### ISSUE THREAD

**Issue by moly** [NuGet/Home#10074]
## Details about Problem

NuGet product used (NuGet.exe | VS UI | Package Manager Console | dotnet.exe): **dotnet.exe**

NuGet version (x.x.x.xxx): **5.7.0.7**

dotnet.exe --version (if appropriate): **3.1.402**

VS version (if appropriate):

OS version (i.e. win10 v1607 (14393.321)): **win10 v1809 (17763.1397)**

Worked before? If so, with which NuGet version:

## Detailed repro steps so we can see the same problem

1. Create a new solution.

2. Create a console project and a class library project and add them to the solution.

3. Add a reference to the class library project from the console project.

4. Create a `Directory.Build.props` at the root of the workspace enabling centrally managed packages.

```
<Project>
  <PropertyGroup>
    <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
  </PropertyGroup>
</Project>
```

5. Create a `Directory.Packages.props` at the root of the workspace that includes one package.

```
<Project>
  <ItemGroup>
    <PackageVersion Include="System.Text.Json" Version="4.7.2" />
  </ItemGroup>
</Project>
```

6. Add that package to the console project only.

```
<Project Sdk="Microsoft.NET.Sdk">

  <ItemGroup>
    <ProjectReference Include="..\ClassProject\ClassProject.csproj" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="System.Text.Json" />
  </ItemGroup>
  
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.1</TargetFramework>
  </PropertyGroup>

</Project>
```

7. Run `dotnet restore --use-lock-file`.

8. Run `dotnet restore --locked-mode`.

9. See error message

```
  Determining projects to restore...
C:\Program Files\dotnet\sdk\3.1.402\NuGet.targets(128,5): error NU1004: The packages lock file is inconsistent with the project dependencies so restore can't be run in locked mode. Disable the RestoreLockedMode MSBuild property or pass an explicit --force-evaluate option to run restore to update the lock file. [C:\Users\user\source\repos\git\repro\repro.sln]
  Failed to restore C:\Users\user\source\repos\git\repro\src\ConsoleProject\ConsoleProject.csproj (in 91 ms).
  Restored C:\Users\user\source\repos\git\repro\src\ClassProject\ClassProject.csproj (in 279 ms).
```

## Other suggested things

The error seems to be that the console project's `packages.lock.json` is wrongly listing all packages in `Directory.Packages.props` as dependencies of the class project.

```json
{
  "version": 2,
  "dependencies": {
    ".NETCoreApp,Version=v3.1": {
      "System.Text.Json": {
        "type": "Direct",
        "requested": "[4.7.2, )",
        "resolved": "4.7.2",
        "contentHash": "TcMd95wcrubm9nHvJEQs70rC0H/8omiSGGpU4FQ/ZA1URIqD4pjmFJh2Mfv1yH1eHgJDWTi2hMDXwTET+zOOyg=="
      },
      "classproject": {
        "type": "Project",
        "dependencies": {
          "System.Text.Json": "4.7.2" # <-- not a dependency of classproject
        }
      }
    }
  }
}
```

If I manually delete the `System.Text.Json` line from the `packages.lock.json` file, `dotnet restore --locked-mode` now works.

### Verbose Logs

Please include verbose logs (NuGet.exe <COMMAND> -verbosity detailed | dotnet.exe <COMMAND> --verbosity diag | etc...)

### Sample Project

https://github.com/moly/lockedfilesissue

**Comment by nkolev92** [comment]
Thanks @moly. 

I was able to repro this both by following your steps and sample project. 

cc @cristinamanum

**Comment by EugeneKrapivin** [comment]
I've just researched the same issue in my projects trying to do exactly the same thing.

**Comment by martinrrm** [comment]
Closing, unable to repro in latest version of Nuget, bug was fixed in Nuget v5.9.1
